server/src/test/java/org/opensearch/common/settings/SettingsModuleTests.java

```diff
@@ -237,4 +237,28 @@ public void testOldMaxClauseCountSetting() {
             ex.getMessage()
         );
     }
+
+    public void testDynamicNodeSettingsRegistration() {
+        Settings settings = Settings.builder().put("some.custom.setting", "2.0").build();
+        SettingsModule module = new SettingsModule(settings, Setting.floatSetting("some.custom.setting", 1.0f, Property.NodeScope));
+        assertNotNull(module.getClusterSettings().get("some.custom.setting"));
+        // For unregistered setting the value is expected to be null
+        assertNull(module.getClusterSettings().get("some.custom.setting2"));
+        assertInstanceBinding(module, Settings.class, (s) -> s == settings);
+
+        module.registerDynamicSetting(Setting.floatSetting("some.custom.setting2", 1.0f, Property.NodeScope));
+        assertNotNull(module.getClusterSettings().get("some.custom.setting2"));
+    }
+
+    public void testDynamicIndexSettingsRegistration() {
+        Settings settings = Settings.builder().put("some.custom.setting", "2.0").build();
+        SettingsModule module = new SettingsModule(settings, Setting.floatSetting("some.custom.setting", 1.0f, Property.NodeScope));
```

Comment:
Should this be `Property.IndexScope`?

Reply:
SettingsModule does not take in index level settings, as it would only take in node level settings and will throw exceptions. 

https://github.com/opensearch-project/OpenSearch/blob/10bff0c9f5b9ca78b3dc50f5c704dabf41b9d535/server/src/main/java/org/opensearch/common/settings/SettingsModule.java#L121

Reply:
I trust you know what you're doing here. My pattern-matching spidey-senses just noticed the inconsistency with Node here and Index everywhere else in this method.

Reply:
Haha yeah, that was intended :)